pkg/controller/certificate-shim/helper.go

```diff
@@ -94,3 +142,26 @@ func translateAnnotations(crt *cmapi.Certificate, ingLikeAnnotations map[string]
 	}
 	return nil
 }
+
+// splitWithEscapeCSV parses the given input as a single line of CSV, which allows
+// a comma-separated list of strings to be parsed while allowing commas to be present
+// in each field. For example, a user can specify:
+// "10 Downing Street, Westminster",Manchester
+// to produce []string{"10 Downing Street, Westminster", "Manchester"}, keeping the comma
+// in the first address. Empty lines or multiple CSV records are both rejected.
+func splitWithEscapeCSV(in string) ([]string, error) {
+	reader := csv.NewReader(strings.NewReader(in))
+
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %q as CSV: %w", in, err)
+	}
+
+	if len(records) == 0 {
+		return nil, fmt.Errorf("no values found after parsing %q", in)
+	} else if len(records) > 1 {
+		return nil, fmt.Errorf("refusing to use %q as input as it parses as multiple lines of CSV", in)
+	}
+
+	return records[0], nil
+}
```

Comment:
suggestion: I think this function would benefit from a few explicit tests, in `pkg/controller/certificate-shim/helper_test.go`

This'll help with debugging and with confidence that it works. Any of the examples I've given in other comments would make good test cases.